### PR TITLE
frontend - most components, basic api integration, designed layout 

### PR DESCRIPTION
**Pull request by otomer**
first frontend PR: (it's 48 files, but don't get scared, there are some assets, deleted files and component's responsibility separation)

**items:**
- adding assets (images)
- creating most of the components
- first integration with backend api (axios) to retrieve only restaurants and cuisines.
- designed layout

screen shot 2019-01-06 at 12 42 36

**what's next:**
- integrate redux and move api requests to actions
- ui components behaviours such as: search, filters, etc.
- map integration
- showing a real promoted restaurant (was hard coded)

### CODE REVIEW

File: frontend/src/components/DeliveryTime/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if (n < 60) {
+      return 'fast';
+    } else if (n < 120) {
+      return 'fair';
+    }
+    return 'slow';
+  }
+
+  return (
+    <p className={'delivery ' + getTimeLabel(time)}>
```
Comment (GeReV) [inline frontend/src/components/DeliveryTime/index.js:20]: Check out a package called [`classnames`](https://github.com/JedWatson/classnames).

File: frontend/src/components/Footer/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import * as Constants from '../../constants';
+import './footer.scss';
+
+const Footer = () => {
+  return (
+    <footer className="footer-container">
+      Copyright © 2019 {Constants.APP_NAME}. All rights reserved
```
Comment (GeReV) [inline frontend/src/components/Footer/index.js:8]: © == `&copy;` 😛

File: frontend/src/components/ListingSection/listing-section.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  padding-bottom: 1rem;
+  // color: var(--we-grey-DETROIT);
+  max-height: 1000px;
+  font-size: 13px;
+}
+
+.flexbox-container {
+  display: -ms-flex;
+  display: -webkit-flex;
+  display: flex;
```
Comment (GeReV) [inline frontend/src/components/ListingSection/listing-section.scss:13]: The Flexbox layout model is pretty much everywhere right now: https://www.caniuse.com/#feat=flexbox
Probably no need to support really old browsers.

File: frontend/src/components/Rating/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+const Rating = props => {
+  const { rating } = props;
+  const roundedRating = Math.round(props.rating);
+  return (
+    <span className="rating" title={rating}>
+      {Array.apply(null, { length: 3 })
+        .map(Number.call, Number)
+        .map((x, i) => (
+          <i
```
Comment (GeReV) [inline frontend/src/components/Rating/index.js:13]: Are you sure this won't collide with the variable `i`?

File: frontend/src/components/Rating/index.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import PropTypes from 'prop-types';
+import './rating.scss';
+
+const Rating = props => {
+  const { rating } = props;
+  const roundedRating = Math.round(props.rating);
+  return (
+    <span className="rating" title={rating}>
+      {Array.apply(null, { length: 3 })
+        .map(Number.call, Number)
```
Comment (GeReV) [inline frontend/src/components/Rating/index.js:11]: That's a really long way of saying `[0, 1, 2]`...

File: frontend/src/containers/CategoriesContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import CuisinesNavigation from '../../components/CuisinesNavigation';
+
+export default class CategoriesContainer extends Component {
+  state = {
+    cuisines: [],
+  };
+
+  componentDidMount() {
+    // TODO: after adding redux - set this as an action
+    axios.get('cuisines').then(res => {
```
Comment (GeReV) [inline frontend/src/containers/CategoriesContainer/index.js:12]: You could just use `window.fetch`.

File: frontend/src/helpers/time-helper.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const formatAsHoursMinutes = minutes => {
+  const num = minutes;
+  const hours = num / 60;
+  const floorHours = Math.floor(hours);
+  const min = (hours - floorHours) * 60;
+  const roundMinutes = Math.round(min);
+  return floorHours + ' hour(s) and ' + roundMinutes + ' minute(s)';
```
Comment (GeReV) [inline frontend/src/helpers/time-helper.js:7]: You can use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

**Review by GeReV — APPROVED** [review]

File: frontend/src/components/Rating/index.js (hunk 5 above)
Comment (otomer) [reply frontend/src/components/Rating/index.js:11]: I agree, tried to be over dynamic, kinda :)

File: frontend/src/components/Footer/index.js (hunk 2 above)
Comment (otomer) [reply frontend/src/components/Footer/index.js:8]: Gotta defend my copyrights

File: frontend/src/components/DeliveryTime/index.js (hunk 1 above)
Comment (otomer) [reply frontend/src/components/DeliveryTime/index.js:20]: Great known package, definitely using it.

**Pull request merged**